src/vconsole/vconsole-setup.c

```diff
@@ -298,14 +298,28 @@ static void setup_remaining_vcs(int src_fd, unsigned src_idx, bool utf8) {
 
                 r = ioctl(fd_d, KDFONTOP, &cfo);
                 if (r < 0) {
-                        log_warning_errno(errno, "KD_FONT_OP_SET failed, fonts will not be copied to tty%u: %m", i);
+                        int last_errno, mode;
+
+                        /* The fonts couldn't have been copied. It might be due to the
+                         * terminal being in graphical mode. In this case the kernel
+                         * returns -EINVAL which is too generic for distinguishing this
+                         * specific case. So we need to retrieve the terminal mode and if
+                         * the graphical mode is in used, let's assume that something else
+                         * is using the terminal and the failure was expected as we
+                         * shouldn't have tried to copy the fonts. */
+
+                        last_errno = errno;
+                        if (ioctl(fd_d, KDGETMODE, &mode) >= 0 && mode != KD_TEXT)
+                                last_errno = -EBUSY;
+
+                        log_full_errno(last_errno == -EBUSY ? LOG_DEBUG : LOG_WARNING, last_errno,
+                                       "KD_FONT_OP_SET failed, fonts will not be copied to tty%u: %m", i);
```

Comment:
So... `last_errno` is either a *postive* errno value, or a negative -EBUSY. Also, the synthethic -EBUSY is logged and returned. 
This will work fine, but somehow it feels wrong. I think that the original errno should be returned (even if it is always -EINVAL now, but the kernel could change). So what about using a separate boolean for `ioctl(fd_d, KDGETMODE, &mode) >= 0 && mode != KD_TEXT` instead?

Reply:
If we do that it won't be obvious that the font copy failed because the terminal is busy.